google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -293,13 +282,25 @@ private T performSet(
       write.setUpdateMask(documentMask.toPb());
     }
 
-    writes.add(new WriteOperation(documentReference, write));
+    return writesAdd(new WriteOperation(documentReference, write));
+  }
 
-    return wrapResult(writes.size() - 1);
+  private T writesAdd(WriteOperation operation) {
```

Comment:
nit: Consider renaming `writesAdd()` to `addWrite()`, which, IMO, is a more conventional name. Also, consider having this method take the `DocumentReference` and `Write` as an argument to deduplicate creating of the `WriteOperation` at the call sites.

Reply:
Done